prow/github/client.go

```diff
@@ -1257,6 +1261,23 @@ func (c *client) authHeader() string {
 	return fmt.Sprintf("Bearer %s", token)
 }
 
+func (c *client) apiVersion() string {
+	resp, err := http.Get("https://api.github.com/versions")
```

Comment:
We totally do not want to do such `GET` on all calls by the client... The version validity check is a neat idea but it should be done just once (at the time of client creation, I guess), through the client call chain itself (so that we get various logging, retry logic etc) and with proper error handling.

I think it would be totally acceptable for this PR to simply hardcode `2022-11-28` (the only valid version at this time) into a const and send a header with that. That would declare "prow client is using specific API version" which IMO totally fine. I'd probably also added a log message to the client creation that would log the version used.

You could invest time into making the API version configurable by e.g. a constructor taking a specific api version (then you could also include a method for verifying that the version is supported), but I think that at time we'd also want to actually think how different API versions would be supported in the client (=how to actually deal with differences?). At this point in time, that's not needed, and we also do not know whether the new API versions will even be relevant to Prow (we do not know how GitHub actually plans to break their APIs :) ), so I'm inclining towards calling [YAGNI](https://en.wikipedia.org/wiki/You_aren%27t_gonna_need_it) and vote for just harcoding the version now.